lib/Sema/CheckedCAlias.cpp

```diff
@@ -333,3 +336,300 @@ void Sema::TrackMemberBoundsDependences(FieldDecl *FD, BoundsExpr *BE) {
   if (BE)
     CollectBoundsMemberUses(FD, getASTContext()).TraverseStmt(BE);
 }
+
+namespace {
+// Update map from variables to bounds expressions that use those variables.
+class UpdateDependences : public RecursiveASTVisitor<UpdateDependences> {
+private:
+   bool Add;              // whether to add or remove dependence.
+   VarDecl *BoundsDecl;   // variable with bounds declaration.
+   Sema::BoundsDependencyTracker::DependentMap &Map;
+
+   void AddDependence(VarDecl *D) {
+    if (D == BoundsDecl)   // Do not add self-dependences.
```

Comment:
Just a personal view, I prefer letting constant variables be left values, like:
BoundsDecl == D;
By doing this we can avoid the equal sing missing bug, buggy code like D = BoundsDecl will never appear again.